NickvisionTubeConverter.Shared/Models/VideoUrlInfo.cs

```diff
@@ -153,4 +152,17 @@ private VideoUrlInfo(string url)
             }
         });
     }
+
+    /// <summary>
+    /// Parses video info from the Python dictionary returned by yt-dlp
+    /// </summary>
+    /// <param name="videoInfo">The dictionary</param>
+    /// <param name="isPartOfPlaylist">Whether or not the video is part of a playlist</param>
+    /// <returns>A VideoUrlInfo object</returns>
+    private static VideoInfo ParseVideoInfo(Python.Runtime.PyDict videoInfo, bool isPartOfPlaylist = false)
```

Comment:
Could you make this method a static method of VideoInfo? Something like:

internal static VideoInfo ParseFromPyDict(...)

Then you can call it using VideoInfo.ParseFromPyDict() .